ci/cloudbuild/builds/lib/vcpkg.sh

```diff
@@ -24,15 +24,19 @@ source module ci/lib/io.sh
 
 TIMEFORMAT="==> 🕑 vcpkg installed in %R seconds"
 time {
-  VCPKG_RELEASE_VERSION="2021.05.12"
+  VCPKG_RELEASE_VERSION="6e024e744e7717c06ddacd5089401109c6298553"
   VCPKG_ROOT_DIR="${HOME}/vcpkg-${VCPKG_RELEASE_VERSION}"
+  WORKSPACE=$(pwd)
   io::log_h2 "Installing vcpkg ${VCPKG_RELEASE_VERSION} -> ${VCPKG_ROOT_DIR}"
   if [[ ! -d "${VCPKG_ROOT_DIR}" ]]; then
     mkdir -p "${VCPKG_ROOT_DIR}"
     # vcpkg needs git history to support versioning, so we clone a recent
     # release tag rather than just extracting a tarball without history.
-    git clone https://github.com/microsoft/vcpkg.git "${VCPKG_ROOT_DIR}" \
-      -b "${VCPKG_RELEASE_VERSION}"
+    git clone https://github.com/microsoft/vcpkg.git "${VCPKG_ROOT_DIR}"
+    cd "${VCPKG_ROOT_DIR}" || exit
+    git checkout "${VCPKG_RELEASE_VERSION}"
+    cd "${WORKSPACE}" || exit
```

Comment:
Can `clone`'s `--branch` option do that checkout?

Reply:
--branch requires the argument to actually be a branch. It won't work with just a commit hash.